### PR TITLE
feat: add sync command [rules engine only]

### PR DESCRIPTION
**Pull request by PatrickMenoti**
**WHAT**
- Add sync command:
  - This will check if there are rules engine created remotely that are not found in local azion.json file, and add them. 

TO DO [Future PRs]:
- Add sync to other resources;
- Add this flow to deploy command;

### CODE REVIEW

File: messages/sync/messages.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package sync
+
+const (
+	USAGE            = "sync"
+	SHORTDESCRIPTION = "Synchronizes local azion.json file with remote resources"
```
Comment (gabriel-azion) [inline messages/sync/messages.go:5]:
```suggestion
	SHORTDESCRIPTION = "Synchronizes the local azion.json file with remote resources"
```

**Review by gabriel-azion — APPROVED** [review]

File: pkg/cmd/sync/tasks.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+	client := api.NewClient(f.HttpClient, f.Config.GetString("api_url"), f.Config.GetString("token"))
+	resp, err := client.ListRulesEngine(C, Opts, info.Conf.Application.ID, "request")
+	if err != nil {
+		return err
+	}
+
+	for _, rule := range resp.Results {
+		if id := info.RuleIds[rule.Name]; id > 0 || rule.Name == "Default Rule" {
+			// if remote rule is also on local environment, no action is needed
```
Comment (PabloDiehl) [inline pkg/cmd/sync/tasks.go:43]: Considering we're only checking the ID... What if something in the rule was changed, like the behavior for example? 🤔

**Review by PabloDiehl — APPROVED** [review]

File: pkg/cmd/sync/tasks.go (hunk 2 above)
Comment (PatrickMenoti) [reply pkg/cmd/sync/tasks.go:43]: This command is used for another purpose:
- We check if there are any rules that were CREATED remotely, but not found on your local azion.json file. It does not consider any change in behavior in existing rules

Comment (PabloDiehl) [reply pkg/cmd/sync/tasks.go:43]: ata

**Review by maxwelbm — APPROVED** [review]

**Pull request merged**